malaria24/ona/admin.py

```diff
@@ -222,8 +232,28 @@ def upload_facility_codes_view(self, request):
         if request.method == 'POST':
             form = FacilityUploadForm(request.POST, request.FILES)
             if form.is_valid():
+                file = form.cleaned_data['upload'].read().decode('utf-8')
+
+                try:
+                    file_content_json = json.loads(file)
+                except json.JSONDecodeError:
+                    string_io_file = io.StringIO(file)
+                    next(string_io_file, None)
+                    file_content_json = list(csv.DictReader(
```

Comment:
Here it seems like we're throwing away the first row of the CSV (presumably it's the headers), and then reading in the rest of the rows, with the column names that we expect.

What might be better to do instead is, to leave out the `fieldnames` parameter to `DictReader`. What that will do is read the first row, assume that it's the field names, and then read the rest of the rows as data. The advantage of this is that the columns can be in a different order, or there could be extra columns, but as long as the column names are correct, we'll be able to process them.

Reply:
Initially I did try to do it this way but I was getting very long strange OrderedDict objects with the key "F" and the values were each individual character in the file, which was wild. I didn't actually go back and check if it worked without the values, but I will quickly do that now :)